src/controller/controller.ts

```diff
@@ -184,6 +184,18 @@ class Controller extends events.EventEmitter {
             databaseCoordinator.changeIeeeAddress(coordinator.ieeeAddr);
         }
 
+        if (startResult === 'resumed') {
+            const netParams = (await this.getNetworkParameters());
+
+            if (this.options.network.channelList[0] !== netParams.channel) {
+                debug.log(`Scheduling switch channel broadcast in 3min.`);
+                // NOTE: stacks that won't support channel switching will not report as 'resumed', they'll have 'reset'/'restored'
```

Comment:
Is this true? E.g. the deconz adapter also returns `resumed`

https://github.com/Koenkk/zigbee-herdsman/blob/85447fbc839fd5d0b42fc7708c2b61486368b3f8/src/adapter/deconz/adapter/deconzAdapter.ts#L98

Reply:
I might have assumed too much here. Since switching without re-pair was not supported, I assumed the combo `'resumed' + different channel` would not happen currently. I'll double check.

Reply:
- deconz:
  - https://github.com/Nerivec/zigbee-herdsman/blob/channel-switching/src/controller/controller.ts#L188
  - https://github.com/Nerivec/zigbee-herdsman/blob/channel-switching/src/adapter/deconz/adapter/deconzAdapter.ts#L891
  - Not sure why all that logic is in `getNetworkParameters`, but seems that will short-circuit the channel mismatch logic in Controller. _Although, that `getNetworkParameters` has a variable scope/re-declaration problem?_
- zigate
  - https://github.com/Nerivec/zigbee-herdsman/blob/channel-switching/src/adapter/zigate/adapter/zigateAdapter.ts#L654
  - Should align channel before reaching the end of start, if I'm not mistaken?
- ezsp
  - https://github.com/Nerivec/zigbee-herdsman/blob/channel-switching/src/adapter/ezsp/driver/driver.ts#L294
  - Will result in 'reset'/'restored'.

Reply:
I'm wondering if we can do something more simple here, I was thinking about adding a method to the adapter: `supportChangeChange()`

Reply:
Added. Can probably be removed once all stacks confirm their logic in regards to this.